lib/async/enumerable/methods/transformers/compact.rb

```diff
@@ -7,7 +7,7 @@ module Transformers
         module Compact
           # Async version of compact that returns an Async::Enumerator for chaining
           def compact
-            self.class.new(super, max_fibers: @max_fibers)
+            self.class.new(super, @async_enumerable_config)
```

Comment:
instead of directly accessing `@async_enumerable_config`, let's create a new method on the Enumerable module called `__async_enumerable_config`. that method may still use refer to an instance variable, but making it a method call will keep the instance_variable references contained to one file